mautic/mautic
# Feature new click filters

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" 
| Bug fix?                               |no
| New feature?                           | yes
| Deprecations?                          |no
| BC breaks?                             |no
| Automated tests included?    | yes

#### Description:
This PR is dependent on https://github.com/mautic/mautic/pull/8649 which must be merged first. 

#8649 had code that included filters from #7786 and #7194. They were removed in https://github.com/mautic/mautic/pull/8649/commits/2cd018a59c214fe45d353e82c7dd22d249cc4b51 so they could be introduced by a new PR (this one). 

This introduces the following new segment filters:
1. Unsubscribed SMS - Manual
2. Clicked any link from any email (existed as `URL from email` but uses a new query builder service now)
2. Clicked any link from any email (date)
3. Clicked any link from any text message
4. Clicked any link from any text message (date)

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Test each of the filters above. They should be self explanatory. 
3. For the unsubscribed SMS, the easiest way to test is to simply publish twilio manually in plugin_integration_settings then go to a contact's profile -> Preference Center -> uncheck the Text Messages channel
4. For the others, you'll need to configure URLs in the channel content and send either the email or SMS to the contact then click the link. 
5. Review the functional tests


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9852"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
